Source/loadsave.cpp

```diff
@@ -1666,10 +1700,10 @@ static void SaveMonster(SaveHelper *file, int i)
 	file->Skip(1); // Alignment
 	file->WriteLE<uint16_t>(pMonster->mExp);
 
-	file->WriteLE<uint8_t>(pMonster->mHit < UINT8_MAX ? pMonster->mHit : UINT8_MAX); // For backwards compatibility
+	file->WriteLE<uint8_t>(std::min<uint16_t>(pMonster->mHit, std::numeric_limits<uint8_t>::max())); // For backwards compatibility
 	file->WriteLE<uint8_t>(pMonster->mMinDamage);
 	file->WriteLE<uint8_t>(pMonster->mMaxDamage);
-	file->WriteLE<uint8_t>(pMonster->mHit2 < UINT8_MAX ? pMonster->mHit2 : UINT8_MAX); // For backwards compatibility
+	file->WriteLE<uint8_t>(std::min<uint16_t>(pMonster->mHit2, std::numeric_limits<uint8_t>::max())); // For backwards compatibility

```

Comment:
Not happy with this but it avoids comparisons between different types and the downcast is recognised as non-truncating in msvc at least.